pkg/artifactory/datasource/security/datasource_artifactory_permission_target_test.go

```diff
@@ -0,0 +1,110 @@
+	}
+	repoTarget := security.PermissionTargetSection{
+		IncludePatterns: []string{"foo/**"},
+		ExcludePatterns: []string{"bar/**"},
+		Repositories:    []string{"example-repo-local"},
+		Actions:         &actions,
+	}
+	buildTarget := security.PermissionTargetSection{
+		IncludePatterns: []string{"foo/**"},
+		ExcludePatterns: []string{"bar/**"},
+		Repositories:    []string{"artifactory-build-info"},
+		Actions:         nil,
+	}
+	releaseBundleTarget := security.PermissionTargetSection{
+		IncludePatterns: []string{"foo/**"},
+		ExcludePatterns: []string{"bar/**"},
+		Repositories:    []string{"release-bundles"},
+		Actions:         nil,
+	}
+
+	permissionTarget := security.PermissionTargetParams{
+		Name:          targetName,
+		Repo:          &repoTarget,
+		Build:         &buildTarget,
+		ReleaseBundle: &releaseBundleTarget,
+	}
+
+	restyClient := acctest.GetTestResty(t)
+	if _, err := restyClient.R().AddRetryCondition(repository.Retry400).SetBody(permissionTarget).Post(security.PermissionsEndPoint + permissionTarget.Name); err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("Create permission target #{permissionTarget.Name} done.")
+}
+
+func TestAccDataSourcePermissionTarget_full(t *testing.T) {
+	_, tempFqrn, name := test.MkNames("test-perm", "artifactory_permission_target")
+
+	temp := `
+  data "artifactory_permission_target" "{{ .permission_name }}" {
+    name = "{{ .permission_name }}"
+  }`
+
+	tempStruct := map[string]string{
+		"permission_name": name,
+	}
+
+	fqrn := "data." + tempFqrn
+	userName := "terraform"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			createPermissionTarget(name, userName, t)
+		},
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy: func(state *terraform.State) error {
+			err := deletePermissionTarget(t, name)
+			_ = acctest.DeleteUser(t, userName)
+			return err
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: util.ExecuteTemplate(fqrn, temp, tempStruct),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(fqrn, "name", name),
+					resource.TestCheckResourceAttr(fqrn, "repo.0.actions.0.users.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "repo.0.actions.0.groups.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "repo.0.repositories.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "repo.0.includes_pattern.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "repo.0.excludes_pattern.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "build.0.repositories.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "build.0.includes_pattern.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "build.0.excludes_pattern.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "release_bundle.0.repositories.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "release_bundle.0.includes_pattern.#", "1"),
+					resource.TestCheckResourceAttr(fqrn, "release_bundle.0.excludes_pattern.#", "1"),
```

Comment:
We are only checking the number of items in each of the list, but not the content. Let's verify the content as well.